docs/building-modules.md

```diff
@@ -132,13 +132,80 @@ individual methods in
 
 - [DOMException](https://developer.mozilla.org/en-US/docs/Web/API/DOMException)
 
+### libp2p
+
+Zinnia comes with a built-in libp2p node based on
+[rust-libp2p](https://github.com/libp2p/rust-libp2p). The node is shared by all
+Station Modules running on Zinnia. This way we can keep the number of open
+connections lower and avoid duplicate entries in routing tables.
+
+The initial version comes with a limited subset of features. We will be adding
+more features based on feedback from our users (Station Module builders).
+
+#### Networking stack
+
+- Transport: `tcp` using system DNS resolver
+- Multistream-select V1
+- Authentication: `noise` with `XX` handshake pattern using X25519 DH keys
+- Stream multiplexing: both `yamux` and `mplex`
```

Comment:
In case `mplex` is not needed, I suggest not offering it at all. Yamux is the better multiplexer in all cases.

You might need it in case you want to talk to old js-libp2p implementations.

See https://github.com/libp2p/specs/pull/402 for rational.

Reply:
Thank you for pointing this out! Somehow I thought that the JavaScript version of Yamux was not production ready yet. 🙈

What are the downsides of supporting both `yamux` and `mplex`? Is there a way how to configure libp2p to prefer `yamux` with a fallback to `mplex` where needed?

We want to allow connectivity with as many kinds of peers as reasonably possible. That's why I slightly prefer to support both multiplexing options. On the other hand, I expect that most Zinnia modules will be written for new projects, where the other side will also be created from scratch. Therefore, module builders should be able to tweak their peer config to support yamux.

Reply:
> Is there a way how to configure libp2p to prefer `yamux` with a fallback to `mplex` where needed?

Just make sure to list `yamux` before `mplex` in stream multiplexer setup.

Keeping `mplex` around for interoperability sounds reasonable.

Reply:
Thanks. Here is our configuration, I think we are good.

https://github.com/filecoin-station/zinnia/blob/cc19e97d103c630f19f1ba772e9dc34283259850/ext/libp2p/peer.rs#L189-L192